hips/hip-9999.md

```diff
@@ -0,0 +1,240 @@
+
+In general, allowing chart developers to run jobs and present that feedback directly to the users could also open up additional use cases beyond just the preflight use case that motivated this HIP. I could imagine scenarios where maybe CVE warnings are presented or specific upgrade feedback is presented instead of just helm install failure.
+
+## Rationale
+
+There are other ways that this could be implemented. For example, we could have a separate preflight hook type. However, this new hook type wouldn't be handled at all by previous versions of helm. With this design, it requires minimial changes to helm and allows for backwards compatibility.
+
+Another strategy could be for helm to include Troubleshoot.sh as a dependent library, but this could result in too tight of a coupling between the projects and lower overall flexibility and adaptability. 
+
+## Specification
+
+Templates could include the following annotations on Batch Jobs:
+    
+```yaml
+        "helm.sh/hook": pre-install, pre-upgrade
+        "helm.sh/hook-output-log-policy": hook-failed # or hook-succeeded or hook-failed,hook-succeeded
+```
+
+
+`helm.sh/hook-output-log-policy` would indicate that helm should display the output of the job to the user.
+
+Additionally a new user flag should be created `--no-log-output`  that would skip the output of logs.
+
+## Backwards compatibility
+
+The only backwards compatibility concern would be that scripts parsing `helm install` output would see some additional text in the case of logs being output. The fact that notes already make the output unstructured should mitigate any concern here. Since we already are trusting chart developers to provide output in the form of notes, this is a logical extension of that that allows the developer to provide more dynamic output. 
+
+## Security implications
+
+Potentially the preflight checks could check for security misconfigurations that could enhance the security of the chart deployment.
+
+## How to teach this
+
+In the first instance, documentation plus the help text for `helm install` would explain the feature.
+
+An example template could be provided in documentation showing how to use this feature with a generic command used in a hook.
+
+A more advanced example showing how to use the new feature with Troubleshoot.sh to provide preflight checks could be linked in the documentation, provided directly in the documentation, or provided on the Troubleshoot.sh documentation site independently.
+
+## Reference implementation
+
+The `safe-install` plugin (link in references) demonstrates what running preflights could look like, but not in the fashion implemented in this HIP.
+
```

Comment:
`safe-install` isn't a reference implementation of the feature. The PR that's been started for this would be a reference implementation. Can we describe what would change in helm at a high level and point to the PR.